src/components/common/Map/NewTarget/index.js

```diff
@@ -0,0 +1,23 @@
+import { useHistory } from 'react-router-dom';
+import { useMapEvents } from 'react-leaflet';
+import { setNewTargetCoords } from '../../../../state/actions/targetActions';
+import useDispatch from '../../../../hooks/useDispatch';
+
+const NewTarget = ({ setNewMarker }) => {
+  const history = useHistory();
+  const updateCoordinates = useDispatch(setNewTargetCoords);
+
+  useMapEvents({
+    click(e) {
+      const coordinates = [e.latlng.lat, e.latlng.lng];
+      setNewMarker(coordinates);
+      updateCoordinates(coordinates);
+      history.push('/'); // forces re-render of create target form, to set new coords on each click on the map
```

Comment:
hmm is this a hack or where did you get this?

Reply:
😄 it's a hack. I need to force a re-render of the form so the coordinates that are stored in Redux on each map click are reflected in the corresponding inputs. The way I currently have to pre-set default values in a form is through the `initialValues` property given by the `useForm` hook, and those values are only setted the first time the form is rendered.

Otherwise, the only time `latitude` and `longitude` are correctly setted is when the user clicks on the map for the first time. Afterwards, each click the user does in a different part of the map is not correctly reflected in the form (although yes in Redux).